crypto/sig_abstractions.go

```diff
@@ -105,20 +107,20 @@ func (z *GenericVerifyingKey) IsValid() error {
 // GetSigner fetches the Signer type that is stored inside this GenericSigningKey.
 func (z *GenericSigningKey) GetSigner() Signer {
 	switch z.Type {
-	case DilithiumType:
-		return &z.DilithiumSigner
+	case FalconType:
+		return &z.FalconSigner
 	case Ed25519Type:
 		return &z.Ed25519Singer
 	default:
-		return &invalidSinger{}
+		return NewInvalidSinger()
```

Comment:
we wanted to avoid returning an error while invoking the "get" function. getting the invalidSigner is very unlikely because the check is performed in the MSGPACK code.